modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/DatabaseRegistry.java

```diff
@@ -259,12 +259,12 @@ void retrieveAndUpdateDatabase(String databaseName, GeoIpTaskState.Metadata meta
             bytes -> Files.write(databaseTmpGzFile, bytes, StandardOpenOption.APPEND),
             () -> {
                 LOGGER.debug("decompressing [{}]", databaseTmpGzFile.getFileName());
-                decompress(databaseTmpGzFile, databaseTmpFile);
 
                 Path databaseFile = geoipTmpDirectory.resolve(databaseName);
                 // tarball contains <database_name>.mmdb, LICENSE.txt, COPYRIGHTS.txt and optional README.txt files.
                 // we store mmdb file as is and prepend database name to all other entries to avoid conflicts
-                try (TarInputStream is = new TarInputStream(new BufferedInputStream(Files.newInputStream(databaseTmpFile)))) {
+                try (TarInputStream is =
+                         new TarInputStream(new GZIPInputStream(new BufferedInputStream(Files.newInputStream(databaseTmpGzFile))))) {
```

Comment:
I think you may still want to pass 8192 into the GZIPInputStream constructor, if you want to keep the buffer at 8k? Otherwise the default is 512

Reply:
That makes sense, I've updated it